src/network/providers/provider.ts

```diff
@@ -0,0 +1,141 @@
+import EthereumProvider from './ethereum-provider';
+import SolanaProvider from './solana-provider';
+import {JsonRpcResponse} from 'hardhat/types';
+import {DomainRegistry} from '../../name_service/types';
+import logger from '../../core/log';
+import {RPCRequest} from '../../rpc/rpc-handlers';
+
+const log = logger.child({module: 'BlockchainProvider'});
+
+export enum ChainId {
+    PointYnet = 'ynet',
+    EthereumMainnet = 1,
+    EthereumRinkeby = 4,
+    SolanaMainnet = 'mainnet-beta',
+    SolanaDevnet = 'devnet'
+}
+
+export enum Blockchain {
+    ETHEREUM = 'ethereum',
+    SOLANA = 'solana'
+}
+
+/**
+ * All blockchain providers must comply with this interface.
+ */
+export interface Provider {
+    name(chainId: ChainId): string;
+    resolveDomain(chainId: ChainId, domain: string): Promise<DomainRegistry>;
+    setDomainContent(
+        chainId: ChainId,
+        domain: string,
+        data: string
+    ): Promise<Record<string, string|number|boolean>>;
+    send(chainId: ChainId, data: RPCRequest): Promise<JsonRpcResponse>;
+    sendTransaction(chainId: ChainId, reqId: number, txParams: unknown[]): Promise<JsonRpcResponse>;
+    requestAccounts(chainId: ChainId, reqId: number): Promise<JsonRpcResponse>;
+    getBlockNumber?(chainId: ChainId): Promise<number>;
+}
+
+const chainToBlockchain: {[k in ChainId]: Blockchain} = {
+    [ChainId.PointYnet]: Blockchain.ETHEREUM,
+    [ChainId.EthereumMainnet]: Blockchain.ETHEREUM,
+    [ChainId.EthereumRinkeby]: Blockchain.ETHEREUM,
+    [ChainId.SolanaMainnet]: Blockchain.SOLANA,
+    [ChainId.SolanaDevnet]: Blockchain.SOLANA
+};
+
+export const tldToChain: Record<string, ChainId> = {
+    '.sol': ChainId.SolanaMainnet,
+    '.eth': ChainId.EthereumRinkeby
+};
+
+// Type guard to determine if a given string|number is a valid ChainId
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isChainId(chainId: any): chainId is ChainId {
+    return Object.values(ChainId).includes(chainId);
+}
+
+/**
+ * BlockchainProvider is the abstraction that consolidates and exposes
+ * blockchain functionality and provides a unified solution for working
+ * with different chains such as Ethereum and Solana.
+ */
+class BlockchainProvider implements Provider {
+    private providers: {[key in Blockchain]?: Provider} = {};
```

Comment:
`providers` looks something like:

```
providers: {
    ethereum: EthereumProvider,
    solana: SolanaProvider,
}
```

It doesn't handle chains within the same blockchain protocol, that's left to the specific providers. For example, Mainnet, Rinkeby, Ropsten, are handled within the `EthereumProvider`.